### ISSUE TITLE
[java] SwitchStmtsShouldHaveDefault gives warning on exhaustive switch statement when switch target is "this" (enum)

### ISSUE THREAD

**Issue by davidsmf** [pmd/pmd#3539]
**Affects PMD Version:**  6.38.0

**Rule:**

Please provide the rule name and a link to the rule documentation:

SwitchStmtsShouldHaveDefault

https://pmd.github.io/pmd-6.38.0/pmd_rules_java_bestpractices.html#switchstmtsshouldhavedefault

**Description:**

**Code Sample demonstrating the issue:**

```
package com.example.test;

public enum TestEnum {
    ONE, TWO;

    String thisFails() {
        switch (this) {
            case ONE: return "ONE";
            case TWO: return "TWO";
        }
        return "";
    }
    String thisIsOk(TestEnum e) {
        switch (e) {
            case ONE: return "ONE";
            case TWO: return "TWO";
        }
        return "";
    }
}
```

**Expected outcome:**

PMD reports a violation at line 7, but that's wrong. That's a false positive.

**Running PMD through:** Maven

**Comment by oowekyala** [comment]
Hi and thanks for the report, but I'll be closing this as a duplicate of #2946
